service/address_map.hh

```diff
@@ -181,102 +213,82 @@ class address_map_t : public peering_sharded_service<address_map_t<Clock>> {
         }
     }
 
-    template <std::invocable<address_map_t&> F>
-    void replicate(F f, seastar::compat::source_location l = seastar::compat::source_location::current()) {
-        if (this_shard_id() != 0) {
-            auto msg = format("address_map::{}() called on shard {} != 0",
-                l.function_name(), this_shard_id());
-            on_internal_error(rslog, msg);
-        }
-        if (!_replication_fiber) {
-            return;
-        }
-
-        _replication_fiber = _replication_fiber->then([this, f = std::move(f), l] () -> future<> {
-            try {
-                co_await this->container().invoke_on_others([f] (address_map_t& self) {
-                    f(self);
-                });
-            } catch (...) {
-                rslog.error("address_map_t::replicate (called from {}) failed: {}",
-                            l.function_name(), std::current_exception());
-            }
-        });
-    }
+    void apply_to_all(locator::host_id id, entry_mutation m, seastar::compat::source_location l = seastar::compat::source_location::current());
 
-    void replicate_set_nonexpiring(const locator::host_id& id) {
-        replicate([id] (address_map_t& self) {
-            self.handle_set_nonexpiring(id);
-        });
+public: // Used by replicator
+    static void prepare_apply(cluster_mutation& map, const host_mutation& mf) {
+        map.try_emplace(mf.first, entry_mutation{});
     }
 
-    void replicate_set_expiring(const locator::host_id& id) {
-        replicate([id] (address_map_t& self) {
-            self.handle_set_expiring(id);
-        });
+    // Applies mf to map.
+    // Must not throw after prepare_apply(map, mf).
```

Comment:
There is no call to prepare_apply in the latest version

Reply:
replicator::apply_to_all() calls container_type::prepare_apply()

Reply:
The comment is about external calls (on the interface), not about the implementation.